frb_codegen/src/parser/ty.rs

```diff
@@ -42,11 +42,14 @@ impl<'a> TypeParser<'a> {
 }
 
 /// Generic intermediate representation of a type that can appear inside a function signature.
+#[derive(Debug)]
 pub enum SupportedInnerType {
     /// Path types with up to 1 generic type argument on the final segment. All segments before
     /// the last segment are ignored. The generic type argument must also be a valid
     /// `SupportedInnerType`.
     Path(SupportedPathType),
+    /// Array type
+    Array(Box<Self>, usize),
```

Comment:
When reviewing the code, I am wondering, can we do *not* create SupportedInnerType::Array, but instead do something in the SupportedPathType?